pkg/namespace/namespace.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
I'd add the build directive `// +build go1.10` to ensure that we don't run into https://github.com/golang/go/issues/20676, as we modify a thread state (with `netns.Set(..)`) in this package.

Reply:
We already require Go in version 1.10 to build Cilium, but sure, I can add the directive.